### ISSUE TITLE
[dot-notation] Fix is incorrect when using optional chaining

### ISSUE THREAD

**Issue by alonstern** [typescript-eslint/typescript-eslint#3791]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/dot-notation": ["error"]
  }
}
```

```TS
const value = { key: 'value' }
console.log(value?.['key'])
```

**Expected Result**

Expected the fix to convert it to
```TS
console.log(value?.key)
```

**Actual Result**

The fix converts it to 
```TS
console.log(value?..key)
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.29.2` |
| `@typescript-eslint/parser`        | `4.29.2` |
| `TypeScript`                       | `3.9.3` |
| `ESLint`                           | `6.2.2` |
| `node`                             | `12.19.0` |
